client/js/components/chart.jsx

```diff
@@ -1,11 +1,87 @@
 var React = require("react");
 var AppStore = require("../store/app_store");
+var c3 = require("c3");
+var _ = require("lodash");
 
 var Chart = React.createClass({
   componentDidMount: function() {
-    AppStore.getVisualizationData(this.props.schemaName,
-      AppStore.generateChart
-    );
+    AppStore.renderSchema(this.props.schemaName);
+  },
+  statics: {
+    renderTimeSeriesChart: function(table, headerIndex) {
+      headerIndex = headerIndex || 2;
+      if (Array.isArray(table)) {
+        var csvHeader = AppStore.formatCSVHeader(table[0][headerIndex]);
+        var _schemaCSVData = [csvHeader];
+    
+        for (var i = 1; i < table.length - 1; i++) {
+          _schemaCSVData.push(table[i][headerIndex]);
+        }
+    
+        c3.generate({
+          bindto: ".visualization-view",
+          data: {
+            columns: [_schemaCSVData]
+           },
+          axis: {
+            y: {
+              label: {
+                text: csvHeader,
+                position: "outer-middle"
+              }
+            },
+            x: {
+              show: true,
+              label: {
+                text: "Time",
+                position: "outer-middle"
+              }
+            }
+          }
+        });
+      }
+    },
+    renderHistogramChart: function(csvData) {
+      var transformHistogramData = function(csvData, headerIndex) {
+        var bucketSize = 2;
+        var header = csvData[0][headerIndex];
+        var sortedDataPoints = _.sortBy(_.rest(csvData), function(row) { return parseInt(row[headerIndex]); });
+        var firstPoint = sortedDataPoints[0][headerIndex];
+        var lastPoint = _.last(sortedDataPoints)[headerIndex];
+
+        var buckets = _.range(firstPoint, lastPoint, bucketSize);          
+
+        var pointIndex = 0;
+        return _.reduce(buckets, function(accum, bucket, ind) {
+          accum.push(0);
+          while(sortedDataPoints[pointIndex][headerIndex] < bucket) {
+            accum[ind] += 1;
+            pointIndex += 1;
+          }
+  
+          return accum;
+        }, [header]);
+      };
+
+      c3.generate({
+        bindto: ".visualization-view",
+        data: {
+            //Row Format ['Data Name', num, num, ...]
+            columns: [transformHistogramData(csvData, 2)],
+            type: "bar"
+        },
+        bar: {
+            width: {
+                ratio: 1 // this makes bar width 50% of length between ticks
+            }
+            // or
+            //width: 100 // this makes bar width 100px
```

Comment:
why is this here? what do you think of:

   var width = some_check ? 100 : {ratio: 1}